### PR TITLE
add SE3.transform_from and SE3.transform_to

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/se3.py [hunk 1]
```diff
@@ -284,26 +284,88 @@ def vee(matrix: torch.Tensor) -> torch.Tensor:
...
 
     # only added to avoid casting downstream
     def copy(self, new_name: Optional[str] = None) -> "SE3":
         return cast(SE3, super().copy(new_name=new_name))
 
+    def _transform_shape_check(self, point: Union[Point3, torch.Tensor]):
+        err_msg = "SE3 can only transform 3-D vectors."
+        if isinstance(point, torch.Tensor):
+            if not point.ndim == 2 or point.shape[1] != 3:
+                raise ValueError(err_msg)
```
Comment (mhmukadam) [inline theseus/geometry/se3.py:298]: The error message here might be ambiguous if the user passes a single point without a batch dim.

File: theseus/geometry/se3.py [hunk 2]
```diff
@@ -284,26 +284,88 @@ def vee(matrix: torch.Tensor) -> torch.Tensor:
...
+            Jg[:, 1, 1] = -1
+            Jg[:, 2, 2] = -1
+            Jg[:, 0, 4] = -ret[:, 2]
+            Jg[:, 1, 3] = ret[:, 2]
+            Jg[:, 0, 5] = ret[:, 1]
+            Jg[:, 2, 3] = -ret[:, 1]
+            Jg[:, 1, 5] = -ret[:, 0]
+            Jg[:, 2, 4] = ret[:, 0]
+            # Jacobians for point
+            Jpnt = self[:, :, :3].transpose(1, 2).expand(batch_size, 3, 3)
```
[the remark this answers is not in the document]
Comment (mhmukadam) [reply theseus/geometry/se3.py:367]: Is this still relevant given you added automatic Jacobians in follow up PRs?

[the remark this answers is not in the document]
Comment (mhmukadam) [reply theseus/geometry/se3.py:367]: Do we have a preference or do we want to let users pick which one they want (since you added support for automatic Jacobians in follow up PRs)?

**Review by mhmukadam — APPROVED** [review]

File: theseus/geometry/se3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/se3.py:367]: As discussed offline,  for `AutodiffCostFunction` we actually want sparse jacobians for efficiency, but we'll need to look into using `vmap` to replace the current implementation using `torch.autograd.jacobian` (or some other feasible alternative). This is not an issue when doing `loss.backward()`.

File: theseus/geometry/se3.py (hunk 1 above)
Comment (luisenp) [reply theseus/geometry/se3.py:298]: Let's change the message for 

```python
err_msg = f"SE3 can only transform vectors of shape [batch_size, 3], but input has shape {point.shape}."
```

File: theseus/geometry/se3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/se3.py:367]: I would like to keep `LieGroup.project()` as it is since this is a general formation to project `jacobian` and `gradient` from the `Euclidean space` to `Lie group tangent space`. Instead, we can write another function to handle automatic `sparse jacobian` calculation.

[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/se3.py:367]: Please refer to #77.

[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/se3.py:367]: To be clear, I was referring to eventually changing the implementation of `AutodiffCostFunction` to avoid computing dense gradients, but I don't anticipate doing that would require any changes to `project()`.

[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/se3.py:367]: Agreed.

File: theseus/geometry/se3.py (hunk 1 above)
Comment (fantaosha) [reply theseus/geometry/se3.py:302]: Fixed.
